File: https:/github.com/rajesh-kayal-dev/Leetcode.git/20-ValidParentheses/20-ValidParentheses.js
// Last updated: 6/15/2026, 10:03:34 AM
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {

    const stack =[];
    const map = {
        ')':'(',
        '}':'{',
        ']':'[',
    }

    for(const c of s){

        if(map[c]){
            const topElement = stack.pop();

            if(topElement !== map[c]){
                return false;
            }
        }else{
                stack.push(c);
            }
    }
    
    return stack.length == 0;
    
};

File: https:/github.com/rajesh-kayal-dev/Leetcode.git/20-ValidParentheses/20-ValidParentheses.java
// Last updated: 6/15/2026, 10:03:28 AM
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();

        for(char ch : s.toCharArray()){
            if(ch == '(' || ch == '[' || ch == '{'){
                stack.push(ch);
            }else{
                if(stack.isEmpty()){
                    return false;
                }

                char top = stack.pop();
                if(ch == ')' && top != '('){
                    return false;
                }
                if(ch == ']' && top != '['){
                    return false;
                }
                if(ch == '}' && top != '{'){
                    return false;
                }
            }
        }
        return stack.isEmpty();
    }
}